retrofit/src/main/java/retrofit2/RequestBuilder.java

```diff
@@ -138,6 +134,17 @@ private static void canonicalize(Buffer out, String input, int pos, int limit,
     }
   }
 
+  private static boolean shouldSkipCodePoint(final int codePoint) {
```

Comment:
Is it possible to name this something like `isUnsafeCodePoint` since these are skipped to prevent xss right?  Just a suggestion